### PR TITLE
Revert "avoid using same port number for test (#4147)"

### LINKED ISSUE #4467: flaky test: netTcpListenClose
https://github.com/denoland/deno/runs/527902802
```
failures:

netTcpListenClose
AddrInUse: Address already in use (os error 98)
    at unwrapResponse ($deno$/ops/dispatch_json.ts:43:11)
    at Object.sendSync ($deno$/ops/dispatch_json.ts:72:10)
    at Object.listen ($deno$/ops/net.ts:54:10)
    at Object.listen ($deno$/net.ts:168:22)
    at netTcpListenClose (file:///home/runner/work/deno/deno/cli/js/tests/net_test.ts:12:25)
    at asyncOpSanitizer ($deno$/testing.ts:29:11)
    at resourceSanitizer ($deno$/testing.ts:54:11)
    at TestApi.[Symbol.asyncIterator] ($deno$/testing.ts:222:17)
    at TestApi.next (<anonymous>)
    at Object.runTests ($deno$/testing.ts:396:20)
```

probably related to 60cee4f045778777a16b6fffd6d5b9a1400d7246

### PR DESCRIPTION
**Pull request by bartlomieju**
This reverts commit 60cee4f045778777a16b6fffd6d5b9a1400d7246.

Closes #4467

CC @keroxp

### CODE REVIEW

**Comment by keroxp** [comment]
It's strange to get `AddrInUse`  even though using different ports for each test...other processes in Github Actions use one of those port sometimes?

**Comment by bartlomieju** [comment]
@keroxp there are multiple `deno` processes launched when running tests so there's a high chance that two of them will run simultaneously and take first port (in both cases it will be the same port).

**Comment by nayeemrmn** [comment]
@bartlomieju Could we pass the first port to use in the cli for the unit test runner?

**Comment by bartlomieju** [comment]
I don't think that's necessary, before using random ports, there were no problems with `AddrInUse`

**Comment by ry** [comment]
BTW the kernel has the functionality to choose a random unused port. We should use that rather than choosing a random number and hoping that it is unused.

File: std/examples/echo_server.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 const hostname = "0.0.0.0";
-const port = +(Deno.args[0] ?? "8080");
+const port = 8080;
```
Comment (keroxp) [inline std/examples/echo_server.ts]: @bartlomieju Would you limit port range around 4500?

File: std/http/server_test.ts [hunk 2]
```diff
@@ -499,9 +491,8 @@ test({
 test({
   name: "respond error closes connection",
   async fn(): Promise<void> {
-    const port = randomPort();
     const serverRoutine = async (): Promise<void> => {
-      const server = serve(":" + port);
+      const server = serve(":8124");
```
Comment (keroxp) [inline std/http/server_test.ts]: Same above

Comment (ry) [reply std/http/server_test.ts]: Let's revert and do another fix up patch rather than conflate the two. It will be easier to manage the changes.

**Comment by keroxp** [comment]
Anyway very sorry for making test more flaky...

**Comment by ry** [comment]
@keroxp no worries - it happens!

**Pull request merged**